src/lib/profiles/service-directory/ServiceDirectory.cpp

```diff
@@ -358,10 +362,41 @@ WEAVE_ERROR WeaveServiceManager::init(WeaveExchangeManager *aExchangeMgr,
 
     clearCacheState();
 
+#if WEAVE_CONFIG_PERSIST_SERVICE_DIRECTORY
+
+    if (platform::PersistentServiceDirExist(kMsgType_ServiceEndpointResponse))
+    {
+        WeaveLogProgress(ServiceDirectory, "Persistent Service Directory Exist");
+        msgBuf = PacketBuffer::New();
+        uint16_t len;
+
+        err = platform::LoadPersistentServiceDir(msgBuf->Start(),
+                                                 msgBuf->AvailableDataLength(),
+                                                 len,
+                                                 kMsgType_ServiceEndpointResponse);
+        SuccessOrExit(err);
+
+        msgBuf->SetDataLength(len);
+
+        err = unpackPacketBuffer(msgBuf, false);
+        SuccessOrExit(err);
+
+        mCacheState = kServiceMgrState_Resolved;
+
+        WeaveLogProgress(ServiceDirectory, "resolved");
```

Comment:
Add some additional context in this log clarifying that it was fetched from persisted cache.